src/main/java/com/cedarsoftware/util/convert/MapConversions.java

```diff
@@ -245,8 +254,12 @@ static OffsetTime toOffsetTime(Object from, Converter converter) {
     private static final String[] OFFSET_DATE_TIME_PARAMS = new String[] { YEAR, MONTH, DAY, HOUR, MINUTE, SECOND, NANO, OFFSET_HOUR, OFFSET_MINUTE };
     static OffsetDateTime toOffsetDateTime(Object from, Converter converter) {
         Map<?, ?> map = (Map<?, ?>) from;
-        if (map.containsKey(YEAR) && map.containsKey(OFFSET_HOUR)) {
-            ConverterOptions options = converter.getOptions();
+        ConverterOptions options = converter.getOptions();
+        if (map.containsKey(DATE_TIME) && map.containsKey(OFFSET)) {
+            LocalDateTime dateTime = converter.convert(map.get(DATE_TIME), LocalDateTime.class);
+            ZoneOffset zoneOffset = converter.convert(map.get(OFFSET), ZoneOffset.class);
+            return OffsetDateTime.of(dateTime, zoneOffset);
```

Comment:
When this line throws an exception, a very specific DateTimeException is thrown.  We are catching and rethrowing IllegalArgumentException in these cases.